lib/mongoose-datatables.js

```diff
@@ -52,16 +52,31 @@ function dataTablesPlugin (schema, options) {
       query.populate(params.populate)
     }
 
-    query.exec()
-
     return new Promise((resolve, reject) => {
+      // validate formatter
+      // formatters can be a function or a string from a formatter defined on options
+      if (params.formatter && !(
+        typeof params.formatter === 'function' ||
+        (typeof params.formatter === 'string' && options && options.formatters && options.formatters[params.formatter])
+      )){
+        return reject(new Error('Invalid formatter'))
+      }
+
       Promise
         .all([query.exec(), thisSchema.count(find)])
         .then(function (results) {
           var response = {}
           response[dataKey] = results[0]
```

Comment:
Call the formatter function here.

Reply:
Maybe `formatter ? results[0].map(formatter) :  results[0]`

Reply:
Since it needs to check if the formatter is a string or a function. It was better to do it in a if bellow.